base/src/main/java/edu/artic/base/utils/ActivityExtensions.kt

```diff
@@ -83,4 +85,22 @@ val Activity.statusBarHeight: Int
             statusBarHeight = resources.getDimensionPixelSize(resourceId)
         }
         return statusBarHeight
-    }
\ No newline at end of file
+    }
+
+/**
+ * Takes the array of color attributes ids and return array of colors.
+ */
+fun Context.getThemeColors(colors: IntArray): Array<ColorStateList?> {
+    val found: Array<ColorStateList?>
+    var typedArray: TypedArray? = null
+    try {
+        typedArray = this.obtainStyledAttributes(colors)
+        found = arrayOfNulls(colors.size)
+        for (i in 0 until colors.size) {
```

Comment:
could you do `found = (0 until colors.size).map { typedArray.getColorStateList(it) }.toTypedArray()` ?
Also why is this necessary? Why not just set the status bar color to the kind we want?

Reply:
We don't know what color to use until the fragment is added.

Reply:
@agrosner Updated the code. @Cliabhach is right for the second point.